sdcm/sct_config.py

```diff
@@ -145,6 +147,19 @@ def dict_or_str(value):
     raise ValueError('"{}" isn\'t a dict'.format(value))
 
 
+def dict_or_str_or_pydantic(value):
```

Comment:
how pydantic is used in params?

Reply:
Parameters `mgmt_restore_params`, `mgmt_agent_backup_config` are Pydantic models.

Reply:
hmm, I'm just not sure when/how it is used. I think dict_or_str would be enough.

Reply:
no, I've tried this option at first, will try to recall when it failed exactly

Reply:
That's what happens when these parameters type is `dict_or_str` - parameters validation doesn't pass.:
```
Traceback (most recent call last):
  File "/home/mikita/scylla/repos/scylla-cluster-tests/sdcm/sct_config.py", line 2107, in _validate_value
    opt['type'](self.get(opt['name']))
  File "/home/mikita/scylla/repos/scylla-cluster-tests/sdcm/sct_config.py", line 147, in dict_or_str
    raise ValueError('"{}" isn\'t a dict'.format(value))
ValueError: "batch_size=2 parallel=1" isn't a dict
```